File: lib/Models/Permission.dart
class Permission {
  DateTime createdAt;
  DateTime updatedAt;
  String walletPublicKey;
  String targetUsername;
  String permission;
  Permission({
    required this.createdAt,
    required this.updatedAt,
    required this.walletPublicKey,
    required this.targetUsername,
    required this.permission,
  });
}

File: src/types/permission.ts
export type Permission = {
    createdAt?: Date,
    updatedAt?: Date,
    walletPublicKey: string,
    targetUsername: string,
    fullName: string,
    permission: string,
    permissionState: PermissionState,    
}
  
enum PermissionState {Revoked, Modified, Added}